crates/bevy_render/src/rangefinder.rs

```diff
@@ -0,0 +1,53 @@
+use bevy_math::{Mat4, Vec4};
+
+use crate::view::ExtractedView;
+
+/// A distance calculator for the draw order of [`PhaseItem`](crate::render_phase::PhaseItem)s.
+pub struct ViewRangefinder3d {
+    inverse_view_row_2: Vec4,
+}
+
+impl ViewRangefinder3d {
+    /// Creates a rangefinder for a view
+    pub fn from_view(view: &ExtractedView) -> ViewRangefinder3d {
+        let inverse_view_matrix = view.transform.compute_matrix().inverse();
+        ViewRangefinder3d {
+            inverse_view_row_2: inverse_view_matrix.row(2),
+        }
+    }
+
+    /// Calculates the distance, or view-space Z value, for a transform
+    pub fn distance(&self, transform: &Mat4) -> f32 {
```

Comment:
Add an inline annotation on this.

Reply:
```suggestion
    #[inline]
    pub fn distance(&self, transform: &Mat4) -> f32 {
```

Reply:
Wait, why resolved if this hasn’t been added? :)

Reply:
Lol. I was just reviewing as you were changing things. Never mind

Reply:
Phew, I was confused :-).